include/unifex/inplace_stop_token.hpp

```diff
@@ -150,9 +156,12 @@ inline inplace_stop_token inplace_stop_source::get_token() noexcept {
 template <typename F>
 class inplace_stop_callback final : private inplace_stop_callback_base {
  public:
-  explicit inplace_stop_callback(inplace_stop_token token, F&& func) noexcept(
-      std::is_nothrow_move_constructible_v<F>)
-      : inplace_stop_callback_base(token.source_), func_((F &&) func) {
+  template(typename T)
+    (requires convertible_to<T, F>)
+  explicit inplace_stop_callback(inplace_stop_token token, T&& func) noexcept(
+      std::is_nothrow_constructible_v<F, T>)
+      : inplace_stop_callback_base(token.source_, &inplace_stop_callback::execute_impl)
+      , func_((T&&) func) {
```

Comment:
How much of the code in the ctor depends on the template parameter? Can non-dependent bits be hoisted into the base?

Reply:
We could probably extract the body of the constructor to a helper method in the base class.

The main reason to have it inlined here would be to allow directly calling the function without indirection in the case that the stop-token was already ready. But this would be at the expense of code-size (more inlining).

Reply:
Fixed.